ci/dependency-update/Jenkinsfile

```diff
@@ -138,8 +143,7 @@ pipeline {
 				axes {
 					axis {
 						name 'DEPENDENCY_UPDATE_NAME'
-						// NOTE: Remember to update the settings() method above when changing this.
-						values 'jandex3', 'orm5.6', 'orm6.2', 'orm6.3', 'orm6-in-main-code'
+                        values params.UPGRADE_JOB == 'all' ? configurations.findAll { it.value.requiresLocalOrmBuild }.keySet() as String[] : params.UPGRADE_JOB
```

Comment:
Regardless... it seems bold to exclude any of those jobs. Right now it's fine, but imagine if we later have a job in Search 6.2 that tests against, say, ORM 6.5, and ORM 6.5 gets automated maintenance releases as well.

It's likely we will have a `wip/6.2/dependency-upgrade/orm6.5` branch, to adjust a few things (e.g. the Jandex dependencies).

If we don't run tests against that branch regularly, we won't know when a commit merged into Search's `6.2` branch breaks that branch.

Instead of `requiresLocalOrmBuild`, can't we have a boolean `allowNoDependencyUpdate`, which would skip the checks we currently have in the `Update dependency` stage (the one that checks the dependency was indeed updated)?

Reply:
rigth ... we have something like it in the main branch for the Elasticsearch case, where no sources are modified to run the job... Hmmm, so you are saying we want to run a modified Search with the "same" (not updated) version of ORM. mmm ok... 😃. I was looking at these jobs more in terms of making sure that once we upgrade to the new version Search would still work OK. And then if we want to make sure that a Search version can work with a different ORM version than the one in the poms, we'd be testing that as a part of a main build...
anywaaaay 😃  I'll port the parameter from the main branch.

Reply:
> I was looking at these jobs more in terms of making sure that once we upgrade to the new version Search would still work OK

No, they're also used to ensure compatibility with multiple versions of ORM.

> And then if we want to make sure that a Search version can work with a different ORM version than the one in the poms, we'd be testing that as a part of a main build...

Unfortunately it's sometimes more complex than just running integration test with `-Dversion.org.hibernate.orm=something`, e.g. integration test POMs need to be patched to update some other dependency. So... we would end up doing what we currently do in this job.